test/RoaveTest/SecurityAdvisories/VersionTest.php

```diff
@@ -221,6 +221,21 @@ public function greaterThanComparisonVersionsProvider() : array
             ['1-stable.1.1.1.1', '1-stable.1.1.1.1', false, false],
             // 0 strip check
             ['1-stable.1.1.1.1', '1-stable.1.1.1.1.0', false, false],
+            // issue 91
+            ['2.3.2-p2', '2.3.2', true, false],
+            ['2.3.2-alpha1', '2.3.2-beta1', false, true],
+            ['2.3.2-beta1', '2.3.2-rc1', false, true],
+            ['1.3.2-rc1', '1.3.2', false, true],
+            ['2.3.2-rc1', '2.3.2', false, true],
+            ['2.3.2', '2.3.3-alpha1', false, true],
+            ['2.3.2-p2', '2.3.2-alpha1', true, false],
+            // compare with version higher
+            ['2.3.2-p2', '2.3.3', false, true],
+            // compare with version lower
+            ['2.3.2-p2', '2.3.1', true, false],
+            // compare two patches but with additional stability versions like 1-p1 and 1-p2 so the p2 will be greater
+            ['2.3.2-p1', '2.3.2-p2', false, true],
```

Comment:
@Ocramius I do this `p1` vs `p2` comparison here.
Also I have to admit it starts to get out of hand, data providers grow and it is a bit hard to navigate.

Reply:
Yeah, but that's combinatory logic with edge cases 🤷‍♀️

The good thing is that I'm not (currently) aware of further edge cases